py4j-python/src/py4j/java_gateway.py

```diff
@@ -334,7 +334,13 @@ def launch_gateway(port=0, jarpath="", classpath="", javaopts=[],
 
     # Determine which port the server started on (needed to support
     # ephemeral ports)
-    _port = int(proc.stdout.readline())
+    _port = None
+    while _port is None:
+        try:
+            _port = int(proc.stdout.readline())
+        except ValueError:
+            # If a JVM debug port is exposed, its details will be printed first
+            pass
```

Comment:
Not sure if there's a better way of doing this other than consuming just enough stdout to successfully parse the int.